packages/adapter-prisma/src/index.ts

```diff
@@ -25,53 +25,75 @@ import type {
 } from "@auth/core/adapters"
 
 export function PrismaAdapter(
-  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>
+  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>,
+  options?: {
+    accountModel?: string
+    authenticatorModel?: string
+    sessionModel?: string
+    userModel?: string
+    verificationTokenModel?: string
+  }
 ): Adapter {
   const p = prisma as PrismaClient
+  const accountModel =
+    p[(options?.accountModel ?? "account") as keyof PrismaClient]
+  const authenticatorModel =
+    p[(options?.authenticatorModel ?? "authenticator") as keyof PrismaClient]
+  const sessionModel =
+    p[(options?.sessionModel ?? "session") as keyof PrismaClient]
+  const userModel = p[(options?.userModel ?? "user") as keyof PrismaClient]
+  const verificationTokenModel =
+    p[
+      (options?.verificationTokenModel ??
+        "verificationToken") as keyof PrismaClient
+    ]
```

Comment:
```suggestion
  const accountModel = p[
    (options?.accountModel ?? "account") as keyof PrismaClient
  ] as Prisma.AccountDelegate
  const authenticatorModel = p[
    (options?.authenticatorModel ?? "authenticator") as keyof PrismaClient
  ] as Prisma.AuthenticatorDelegate
  const sessionModel = p[
    (options?.sessionModel ?? "session") as keyof PrismaClient
  ] as Prisma.SessionDelegate
  const userModel = p[
    (options?.userModel ?? "user") as keyof PrismaClient
  ] as Prisma.UserDelegate
  const verificationTokenModel = p[
    (options?.verificationTokenModel ??
      "verificationToken") as keyof PrismaClient
  ] as Prisma.VerificationTokenDelegate
```

This would get rid of the TypeScript errors on the schema that is used for the build (and tests). But TypeScript will give errors if you actually use the options for different models. You need to find some kind of generic Delegate.

I would recommend starting with adding a custom schema + tests first and then try to fix the TypeScript issues after